pkg/vsphere/tasks/waiter.go

```diff
@@ -82,3 +89,54 @@ func WaitForResult(ctx context.Context, f func(context.Context) (ResultWaiter, e
 	}
 	return info, nil
 }
+
+func Retry(ctx context.Context, f func(context.Context) (ResultWaiter, error)) (*types.TaskInfo, error) {
+	rand.NewSource(time.Now().UnixNano()) //creates a more unique random
+	var err error
+	var taskInfo *types.TaskInfo
+	backoffFactor := int64(1)
+
+	//setting up timer
+	timer := time.NewTimer(time.Duration(100) * time.Millisecond)
+	if !timer.Stop() {
+		<-timer.C
+	}
+
+	for {
+		taskInfo, err = WaitForResult(ctx, f)
+
+		if err != nil {
+			return nil, err
+		}
+
+		if taskInfo.Error == nil {
+			break
+		}
+
+		if _, ok := taskInfo.Error.Fault.(types.TaskInProgressFault); !ok {
+			break
```

Comment:
Looks like the 2 callers of Retry() ignore the returned TaskInfo and only check the returned error.  So this Error would be swallowed.  At the very least, this taskInfo.Error should be logged here.  Returning the error in this case or having Retry() callers check taskInfo.Error would be better.